File: my-shop-selenium/src/test/java/pages/MyAccountPage.java
package pages;

import data.dto.UserDTO;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import java.util.List;

public class MyAccountPage extends BasePage {

    //region Locators
    private final static By myPersonalInformationButton = By.cssSelector("a[title='Information']");
    private final static By myAddressesButton = By.cssSelector("a[title='Addresses']");
    private final static By logoutButton = By.cssSelector("a.logout");

    private static final By firstNameInput = By.id("firstname");
    private static final By lastNameInput = By.id("lastname");
    private static final By formCreateAlertError = By.cssSelector("div.alert.alert-danger");
    //endregion

    //region Methods
    public static void clickMyPersonalInformationButton(){
        click(myPersonalInformationButton);
    }

    public static void clickMyAddressesButton(){
        click(myAddressesButton);
    }

    public static void accessPage(){
        get("http://www.automationpractice.pl/index.php?controller=my-account");
    }

    public static void fillFirstNameInput(String firstName) {
        sendKeys(firstNameInput, firstName);
    }

    public static void clearFirstNameInput() {
        clear(firstNameInput);
    }

    public static String geValueFirstNameInput() {
        return getValue(firstNameInput);
    }

    public static void fillLastNameInput(String lastName) {
        sendKeys(lastNameInput, lastName);
    }

    public static void clearLastNameInput() {
        clear(lastNameInput);
    }

    public static String geValueLastNameInput() {
        return getValue(lastNameInput);
    }

    public static void clickLogoutButton(){
        click(logoutButton);
    }

    public static List<String> getMeessagesAlertError() {
        return getElement(formCreateAlertError)
                .findElements(By.tagName("li")).stream()
                .map(WebElement::getText)
                .toList();
    }
    //endregion

    //region Fluxos
    public static void fromLoginToMyPersonalInformation(UserDTO user){
        LoginPage.login(user.getEmail(), user.getPassword());
        clickMyPersonalInformationButton();
    }

    public static void fromAnyPageToMyPersonalInformation(){
        accessPage();
        clickMyPersonalInformationButton();
    }

    public static void fromLoginToMyAddresses(UserDTO user){
        LoginPage.login(user.getEmail(), user.getPassword());
        clickMyAddressesButton();
    }

    public static void fromAnyPageToMyAddresses(){
        accessPage();
        clickMyAddressesButton();
    }
    //endregion

}

File: my-shop-cypress/cypress/e2e/pages/my-account/my-account-page.ts
import LoginPage from '../login-page';
import UserDTO from '../../data/dto/user-dto';

export default class MyAccountPage {

    static myPersonalInformationButton = "a[title='Information']";
    static myAddressesButton = "a[title='Addresses']";
    static logoutButton = "a.logout";
    static firstNameInput = "#firstname";
    static lastNameInput = "#lastname";
    static formCreateAlertError = "div.alert.alert-danger li";
    static pageUrl = Cypress.config("baseUrl") + "?controller=my-account";

    static clickMyPersonalInformationButton() {
        cy.get(this.myPersonalInformationButton).click();
    }

    static clickMyAddressesButton() {
        cy.get(this.myAddressesButton).click();
    }

    static accessPage() {
        cy.visit(this.pageUrl);
    }

    static fillFirstNameInput(firstName) {
        cy.get(this.firstNameInput).type(firstName);
    }

    static clearFirstNameInput() {
        cy.get(this.firstNameInput).clear();
    }

    static getValueFirstNameInput() {
        return cy.get(this.firstNameInput).invoke('attr', 'value');
    }

    static fillLastNameInput(lastName) {
        cy.get(this.lastNameInput).type(lastName);
    }

    static clearLastNameInput() {
        cy.get(this.lastNameInput).clear();
    }

    static getValueLastNameInput() {
        return cy.get(this.lastNameInput).invoke('attr', 'value');
    }

    static clickLogoutButton() {
        cy.get(this.logoutButton).click();
    }

    static getMeessagesAlertError() {
        return cy.get(this.formCreateAlertError).invoke('text');
    }

    static async fromLoginToMyPersonalInformation(user: UserDTO) {
        LoginPage.accessPage();
        LoginPage.login(user.email, user.password);
        this.clickMyPersonalInformationButton();
    }

    static async fromAnyPageToMyPersonalInformation() {
        this.accessPage();
        this.clickMyPersonalInformationButton();
    }

    static async fromLoginToMyAddresses(user: UserDTO) {
        LoginPage.accessPage();
        LoginPage.login(user.email, user.password);
        this.clickMyAddressesButton();
    }

    static async fromAnyPageToMyAddresses() {
        this.accessPage();
        this.clickMyAddressesButton();
    }
}
